parachain/pallets/inbound-queue/src/lib.rs

```diff
@@ -117,10 +137,9 @@ pub mod pallet {
 				}
 			})?;
 
-			let message_id = MessageId::new(envelope.account, envelope.nonce);
-			T::MessageDispatch::dispatch(envelope.source, message_id, &envelope.payload);
-
-			<LatestVerifiedBlockNumber<T>>::set(block_number);
+			// Reward relayer from the sovereign account of the destination parachain
+			let dest_account = envelope.dest.into_account_truncating();
+			T::Token::transfer(&who, &dest_account, T::Reward::get(), Preservation::Expendable)?;
 
```

Comment:
Is `LatestVerifiedBlockNumber` safe to remove here? I would assume we still need some scanner logic to find the unsynced messages and it is useful to narrow the scan scope down.